boards/Makefile.include.cortexm_common

```diff
@@ -4,11 +4,21 @@ export TARGET_TRIPLE ?= arm-none-eabi
 # Toolchain prefix, defaults to target triple followed by a dash, you will most likely not need to touch this.
 export PREFIX ?= $(if $(TARGET_TRIPLE),$(TARGET_TRIPLE)-)
 
-# we build all cortex boards with the GNU toolchain
+ifeq ($(BUILD_WITH_CLANG),1)
+# Build with LLVM/Clang
+include $(RIOTBOARD)/Makefile.include.llvm
+else
+# Default: use the GNU/GCC toolchain
 include $(RIOTBOARD)/Makefile.include.gnu
+endif
 
 # define build specific options
-export CFLAGS_CPU   = -mcpu=$(MCPU) -mlittle-endian -mthumb -mno-thumb-interwork $(CFLAGS_FPU)
+export CFLAGS_CPU   = -mcpu=$(MCPU) -mlittle-endian -mthumb -fshort-enums $(CFLAGS_FPU)
```

Comment:
as for `-fshort-enums`:
newlib nano on my machine seems to be built with `-fshort-enums` which causes warnings during linking with RIOT built using Clang. It seems like gcc defaults to using short enums and clang defaults to using word size enums because RIO files built with GCC do not cause any warnings.